include/boost/math/special_functions/detail/hypergeometric_asym.hpp

```diff
@@ -44,7 +44,7 @@
         }
         else
         {
-           e = z > (std::numeric_limits<long long>::max)() ? (std::numeric_limits<long long>::max)() : lltrunc(z, pol);
+           e = z > static_cast<T>((std::numeric_limits<long long>::max)()) ? (std::numeric_limits<long long>::max)() : lltrunc(z, pol);
```

Comment:
What's the warning on this one?

My only (very slight) concern, is that for multiprecision types it is in general cheaper to compare to an integer, than to create a temporary T and compare.  In this particular case I very much doubt that anyone will notice the difference in a hypergeometric calculation, but in general we need to be a bit careful about casting integers to T's just out of habit.

Reply:
```
../../../boost/math/special_functions/detail/hypergeometric_asym.hpp:47:21: warning: implicit conversion from 'long long' to 'double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
           e = z > ((std::numeric_limits<long long>::max)()) ? (std::numeric_limits<long long>::max)() : lltrunc(z, pol);
```

Reply:
I bet there are a few of those, lets ignore it for now and merge - thanks for this!